src/gateway_config_miner.erl

```diff
@@ -0,0 +1,98 @@
+-module(gateway_config_miner).
+
+-behavior(gen_server).
+
+-include("gateway_config.hrl").
+-define(SERVICE, 'helium.local.api').
+
+%% gen_server
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%% api
+-export([
+    status/0,
+    pubkey/0,
+    add_gateway/3
+]).
+
+-record(state, {
+    connection = undefined :: grpc_client:connection() | undefined
+}).
+
+-spec status() -> [tuple()].
+status() ->
+    gen_server:call(?MODULE, status).
+
+-spec pubkey() -> {PubKey :: binary(), OnboardingKey :: binary()}.
+pubkey() ->
+    gen_server:call(?MODULE, pubkey).
+
+-spec add_gateway(
+    Owner :: libp2p_crypto:pubkey_bin(),
+    Payer :: libp2p_crypto:pubkey_bin(),
+    Mode :: full | dataonly | light
+) -> {ok, binary()} | {error, term()}.
+add_gateway(Owner, Payer, Mode) ->
+    gen_server:call(?MODULE, {add_gateway, [Owner, Payer, Mode]}).
+
+init(_) ->
+    erlang:register(?MODULE, self()),
+    {ok, #state{}}.
+
+handle_call(Call, From, State = #state{connection = undefined}) ->
+    case grpc_client:connect(tcp, "localhost", 4467) of
+        {ok, Connection} ->
+            handle_call(Call, From, State#state{connection = Connection});
+        {error, Error} ->
+            {reply, {error, Error}, State}
+    end;
+handle_call(status, _From, State = #state{connection = Connection}) ->
+    case call_unary(Connection, height, #{}) of
+        {ok, Result} ->
+            Height = kvc:path(<<"result.height">>, Result),
```

Comment:
i assume this kvc library holds up okay with a recent OTP since it hasn't been updated since 2016? might be worth pulling it if it's just saving a map pattern match?